state-chain/pallets/cf-swapping/src/tests/fees.rs

```diff
@@ -1045,3 +1045,152 @@ fn gas_calculation_can_handle_extreme_swap_rate() {
 		);
 	});
 }
+
+#[test]
+fn test_get_network_fee() {
+	const REGULAR_NETWORK_FEE: u32 = 5;
+	const INTERNAL_SWAP_NETWORK_FEE: u32 = 6;
+	const MINIMUM_NETWORK_FEE: AssetAmount = 123;
+
+	fn test_get_fee(
+		input_asset_fee: (Asset, Option<u32>),
+		output_asset_fee: (Asset, Option<u32>),
+		is_internal: bool,
+		expected_fee: u32,
+	) {
+		new_test_ext().execute_with(|| {
+			// Set the standard network fee
+			if is_internal {
+				InternalSwapNetworkFee::<Test>::set(FeeRateAndMinimum {
+					rate: Permill::from_percent(INTERNAL_SWAP_NETWORK_FEE),
+					minimum: MINIMUM_NETWORK_FEE,
+				});
+			} else {
+				NetworkFee::<Test>::set(FeeRateAndMinimum {
+					rate: Permill::from_percent(REGULAR_NETWORK_FEE),
+					minimum: MINIMUM_NETWORK_FEE,
+				});
+			}
+
+			// Set the custom network fees for the assets
+			if let (asset, Some(fee)) = input_asset_fee {
+				if is_internal {
+					InternalSwapNetworkFeeForAsset::<Test>::insert(
+						asset,
+						Permill::from_percent(fee),
+					);
+				} else {
+					NetworkFeeForAsset::<Test>::insert(asset, Permill::from_percent(fee));
+				}
+			}
+			if let (asset, Some(fee)) = output_asset_fee {
+				if is_internal {
+					InternalSwapNetworkFeeForAsset::<Test>::insert(
+						asset,
+						Permill::from_percent(fee),
+					);
+				} else {
+					NetworkFeeForAsset::<Test>::insert(asset, Permill::from_percent(fee));
+				}
+			}
+
+			// Get the network fee for the swap
+			let fee = Pallet::<Test>::get_network_fee_for_swap(
+				input_asset_fee.0,
+				output_asset_fee.0,
+				is_internal,
+			);
+
+			// Check that the fee rate and minimum are as expected
+			assert_eq!(fee.minimum, MINIMUM_NETWORK_FEE);
+			assert_eq!(fee.rate, Permill::from_percent(expected_fee));
+		});
+	}
+
+	fn test_all(is_internal: bool) {
+		let network_fee = if is_internal { INTERNAL_SWAP_NETWORK_FEE } else { REGULAR_NETWORK_FEE };
+
+		// The normal network fee is used as fallback when no custom fee is set for either asset.
+		test_get_fee((Asset::Flip, None), (Asset::Eth, None), is_internal, network_fee);
+
+		// Test that the normal network fee is used if one asset is not set
+		test_get_fee(
+			(Asset::Flip, Some(network_fee - 1)),
+			(Asset::Eth, None),
+			is_internal,
+			network_fee,
+		);
+		test_get_fee(
+			(Asset::Flip, None),
+			(Asset::Eth, Some(network_fee - 2)),
+			is_internal,
+			network_fee,
+		);
+
+		// Test that the highest fee is always used
+		test_get_fee((Asset::Flip, Some(20)), (Asset::Eth, Some(25)), is_internal, 25);
+		test_get_fee((Asset::Flip, None), (Asset::Eth, Some(25)), is_internal, 25);
```

Comment:
As discussed, it is confusing that in this case we expect 25, while in the case above (where one of the fees is also missing) we expect the default fee. Migth we worth adding a comment to explain why that is.